test/cql-pytest/test_json.py

```diff
@@ -159,18 +159,12 @@ def test_fromjson_varint_empty_string_prepared(cql, table1):
     with pytest.raises(FunctionFailure):
         cql.execute(stmt, [p, '""'])
 
-# Cassandra allows the strings "true" and "false", not just the JSON constants
-# true and false, to be assigned to a boolean column. However, very strangely,
-# it only allows this for prepared statements, and *not* for unprepared
-# statements - which result in an InvalidRequest!
```

Comment:
This comment about Cassandra's behavior is (unless they fixed it?) still accurate - Cassandra has this bug where it does not NOT allow strings true/false in unprepared statements!
So, your fixed test below will fail on Cassandra (you can easily test it yourself - set the CASSANDRA variable to point to any Cassandra executable you want, and run `test/cql-pytest/run-cassandra test_json.py::test_fromjson_boolean_string_unprepared`). We need to add the "cassandra_bug" fixture to this test to say it is expected to fail on Cassandra, and add to the comment about a few words about the fact that we consider the lack of support for these strings in unprepared statements to be a bug.

Reply:
I launched Cassandra's CCM and the test is passing. In my opinion, the problem was with the test because of the missing `fromJson()` use. Also using cqlsh I am able to insert `"true"` into Cassandra's boolean column and afaik cqlsh statement is unprepared one.

Reply:
Wow, you are absolutely right!! I updated #7915.